bridgesync/processor.go

```diff
@@ -430,6 +430,34 @@ func (p *processor) GetClaims(ctx context.Context, fromBlock, toBlock uint64) ([
 	return claims, nil
 }
 
+func (p *processor) GetClaimByGlobalIndex(ctx context.Context, globalIndex *big.Int) (Claim, error) {
+	if globalIndex == nil {
+		return Claim{}, fmt.Errorf("global index cannot be nil")
+	}
+
+	tx, err := p.startTransaction(ctx, true)
+	if err != nil {
+		return Claim{}, err
+	}
+	defer p.rollbackTransaction(tx)
+
+	var claim Claim
+	if err = meddler.QueryRow(tx, &claim, fmt.Sprintf(`
```

Comment:
We don't need the transactions in case of reading (refer to the #831)
```suggestion
	var claim Claim
	if err = meddler.QueryRow(p.db, &claim, fmt.Sprintf(`
```

Reply:
Fix: https://github.com/agglayer/aggkit/pull/838/commits/a0a491d0e1a9299e71c55ce098802a68184555fe